src/Gateway.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace MyOnlineStore\Omnipay\KlarnaCheckout;
+
+use Klarna\Rest\Transport\Connector;
+use Klarna\Rest\Transport\ConnectorInterface;
+use Omnipay\Common\AbstractGateway;
+
+final class Gateway extends AbstractGateway
+{
+    /**
+     * @inheritDoc
+     */
+    public function getName()
+    {
+        return 'KlarnaCheckout';
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getDefaultParameters()
+    {
+        return [
+            'merchant_id' => '',
+            'secret' => '',
+            'testMode' => true,
+        ];
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function initialize(array $parameters = array())
+    {
+        parent::initialize($parameters);
+
+        $this->parameters->set(
+            'connector',
+            Connector::create(
+                $this->getParameter('merchant_id'),
+                $this->getParameter('secret'),
+                $this->getTestMode() ? ConnectorInterface::EU_TEST_BASE_URL : ConnectorInterface::EU_BASE_URL
```

Comment:
This limits the package to Europe.  EU can be default, but I would suggest an additional parameter to allow NA URLs.